app/assets/javascripts/topology_preview.js

```diff
@@ -1,33 +1,34 @@
 var TopologyPreviewer = (function(){
-  var topologyGraph, svgGroup, data, root, tree, svgHeight, svgWidth;
+  var topologyGraph, presetData, style, svgGroup, data, root, tree, svgHeight, svgWidth,
+      currentStylesheet;
 
   var nodeIds         = 0,
       maxLabelLength  = 20,
-      duration        = 250,
-      endPointCounter = 0,
-      stepSize        = 20,
       ease            = 'cubic-out';
 
   var diagonal = d3.svg.diagonal()
-      .projection(function(d) { return [d.y, d.x]; });
+      .projection(function(d) { return [d.x, d.y]; });
 
   TopologyPreviewer.prototype = {
     preview: function(){
       clear();
 
-      svgHeight = viewerHeight();
-      svgWidth  = viewerWidth();
-      svg       = buildBaseSvg();
-      data      = [graphData()];
-      root      = data[0];
+      currentStylesheet = TopologyStylesheet[style];
```

Comment:
They're defined as a `var` on line 2. Than filled at line 18 until 22. 
They're not globals. See it as this:

````javascript
var A = (function(){
  var b;

  b = 5;
})();

console.log(b);
````

It's a global within the scope of `A` which makes it local. It might also be that I'm confusing your definition of global with my definition of global: http://www.w3schools.com/js/js_scope.asp